client/components/ScenariosList/index.jsx

```diff
@@ -11,38 +12,92 @@ class ScenariosList extends Component {
+            listHeading: '',
+            scenariosToList: []
         };
+
+        this.sortDeletedScenarios = this.sortDeletedScenarios.bind(this);
+        this.getScenariosList = this.getScenariosList.bind(this);
     }
 
     async componentDidMount() {
         await this.props.getScenarios();
+        this.getScenariosList();
     }
 
-    render() {
-        const { category } = this.state;
-        const { scenarios } = this.props;
-        const displayableScenarios =
-            (scenarios &&
-                scenarios.filter(({ categories }) => {
-                    return !category || categories.includes(category);
-                })) ||
-            [];
-
-        // TODO: Expose deleted scenarios these to Admin only
-        // This pushes "deleted" scenarios to the end of the list of Scenarios,
-        // as a temporary means of addressing the display of "deleted"
-        // scenarios.
-        displayableScenarios.forEach((scenario, index, scenarios) => {
+    sortDeletedScenarios(scenarios) {
+        scenarios.forEach((scenario, index, scenarios) => {
             if (scenario.deleted_at) {
                 scenarios.splice(index, 1);
                 scenarios.push(scenario);
             }
         });
 
+        return scenarios;
+    }
+
+    async getScenariosList() {
+        const { category } = this.state;
+        let filteredScenarios = [];
+        let listHeading = '';
+        let authorUsername = '';
+
+        switch (category) {
+            case 'all':
+                filteredScenarios = this.props.scenarios;
+                listHeading = 'All Teacher Moments';
+                break;
+            case 'author':
+                authorUsername = this.props.match.params.username;
+                // Currently we're only showing author views for the current user
+                if (this.props.username !== authorUsername) return;
+
+                listHeading = ` Teacher Moments by ${changeCase.titleCase(
+                    authorUsername
+                )}`;
+                filteredScenarios = this.props.scenarios.filter(
+                    ({ user_is_author }) => {
+                        return user_is_author;
+                    }
+                );
+                break;
+            case 'official':
+            case 'community':
+                filteredScenarios = this.props.scenarios.filter(
+                    ({ categories }) => {
+                        return !category || categories.includes(category);
+                    }
+                );
+                listHeading = `${changeCase.titleCase(
+                    category
+                )} Teacher Moments`;
+                break;
+            case 'completed':
+                // eslint-disable-next-line no-case-declarations
+                const { scenarios } = await (await fetch(
+                    'api/scenarios/run'
+                )).json();
+                filteredScenarios = scenarios;
```

Comment:
`scenarios.filter(scenario => scenario.ended_at !== null)`

Reply:
As we chatted about offline, our plan at the moment is to show all scenarios that have been started _and_ completed. I'll change the language to reflect that it's not just completed moments, unless you'd prefer I just take out the uncompleted moments entirely.

Reply:
I've settled on using the term "continue" over "completed". I think a good future refinement would be to distinguish in the UI the finished and in-progress moments, but I think we can tackle that another time. Let me know what you think of "continue" @rwaldron

Reply:
I love it—"continue" feels more like positive action.